File: core/src/main/java/vn/loitp/flickr/model/photosetgetphotos/Photo.java

package vn.loitp.flickr.model.photosetgetphotos;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Photo {

    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("secret")
    @Expose
    private String secret;
    @SerializedName("server")
    @Expose
    private String server;
    @SerializedName("farm")
    @Expose
    private double farm;
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("isprimary")
    @Expose
    private String isprimary;
    @SerializedName("ispublic")
    @Expose
    private double ispublic;
    @SerializedName("isfriend")
    @Expose
    private double isfriend;
    @SerializedName("isfamily")
    @Expose
    private double isfamily;
    @SerializedName("url_o")
    @Expose
    private String urlO;
    @SerializedName("height_o")
    @Expose
    private String heightO;
    @SerializedName("width_o")
    @Expose
    private String widthO;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSecret() {
        return secret;
    }

    public void setSecret(String secret) {
        this.secret = secret;
    }

    public String getServer() {
        return server;
    }

    public void setServer(String server) {
        this.server = server;
    }

    public double getFarm() {
        return farm;
    }

    public void setFarm(double farm) {
        this.farm = farm;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getIsprimary() {
        return isprimary;
    }

    public void setIsprimary(String isprimary) {
        this.isprimary = isprimary;
    }

    public double getIspublic() {
        return ispublic;
    }

    public void setIspublic(double ispublic) {
        this.ispublic = ispublic;
    }

    public double getIsfriend() {
        return isfriend;
    }

    public void setIsfriend(double isfriend) {
        this.isfriend = isfriend;
    }

    public double getIsfamily() {
        return isfamily;
    }

    public void setIsfamily(double isfamily) {
        this.isfamily = isfamily;
    }

    public String getUrlO() {
        return urlO;
    }

    public void setUrlO(String urlO) {
        this.urlO = urlO;
    }

    public String getHeightO() {
        return heightO;
    }

    public void setHeightO(String heightO) {
        this.heightO = heightO;
    }

    public String getWidthO() {
        return widthO;
    }

    public void setWidthO(String widthO) {
        this.widthO = widthO;
    }

}

File: core/src/main/java/com/loitp/restApi/flickr/model/photoSetGetPhotos/Photo.kt
package com.loitp.restApi.flickr.model.photoSetGetPhotos

import androidx.annotation.Keep
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.loitp.core.base.BaseModel
import com.loitp.core.ext.getFlickrLink100
import com.loitp.core.ext.getFlickrLink1024
import com.loitp.core.ext.getFlickrLink320
import com.loitp.core.ext.getFlickrLink640

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
@Keep
class Photo : BaseModel() {
    @SerializedName("id")
    @Expose
    var id: String? = null

    @SerializedName("secret")
    @Expose
    @Suppress("unused")
    var secret: String? = null

    @SerializedName("server")
    @Expose
    @Suppress("unused")
    var server: String? = null

    @SerializedName("farm")
    @Expose
    @Suppress("unused")
    var farm = 0.0

    @SerializedName("title")
    @Expose
    var title: String = ""

    @SerializedName("isprimary")
    @Expose
    @Suppress("unused")
    var isprimary: String? = null

    @SerializedName("ispublic")
    @Expose
    @Suppress("unused")
    var ispublic = 0.0

    @SerializedName("isfriend")
    @Expose
    @Suppress("unused")
    var isfriend = 0.0

    @SerializedName("isfamily")
    @Expose
    @Suppress("unused")
    var isfamily = 0.0

    @SerializedName("url_o")
    @Expose
    var urlO: String = ""

    @SerializedName("height_o")
    @Expose
    var heightO = 1

    @SerializedName("width_o")
    @Expose
    var widthO = 1

    @SerializedName("url_s")
    @Expose
    var urlS: String = ""

    @SerializedName("height_s")
    @Expose
    @Suppress("unused")
    var heightS = 1

    @SerializedName("width_s")
    @Expose
    @Suppress("unused")
    var widthS = 1

    @SerializedName("url_m")
    @Expose
    var urlM: String = ""

    @SerializedName("height_m")
    @Expose
    @Suppress("unused")
    var heightM = 0

    @SerializedName("width_m")
    @Expose
    @Suppress("unused")
    var widthM = 0

    @Suppress("unused")
    // gif extension have no link large
    val flickrLink100: String
        get() = if (urlO.contains(".gif")) {
            // gif extension have no link large
            urlO
        } else {
            getFlickrLink100(urlM)
        }

    @Suppress("unused")
    // gif extension have no link large
    val flickrLink640: String
        get() = if (urlO.contains(".gif")) {
            // gif extension have no link large
            urlO
        } else {
            getFlickrLink640(urlM)
        }

    // gif extension have no link large
    val flickrLink1024: String
        get() = if (urlO.contains(".gif")) {
            // gif extension have no link large
            urlO
        } else {
            getFlickrLink1024(urlM)
        }

    @Suppress("unused")
    // gif extension have no link large
    val flickrLink320: String
        get() = if (urlO.contains(".gif")) {
            // gif extension have no link large
            urlO
        } else {
            getFlickrLink320(urlM)
        }

    fun calculatorHeight(widthScreen: Int): Int {
        return heightO * widthScreen / widthO
    }

}
